packages/arb-validator-core/message/message.go

```diff
@@ -102,23 +187,41 @@ func UnmarshalFromCheckpoint(msgType MessageType, v value.Value) (InboxMessage,
 		return UnmarshalERC721FromCheckpoint(v)
 	case ContractTransactionType:
 		return UnmarshalContractTransactionFromCheckpoint(v)
+	case TransactionBatchType:
+		return UnmarshalTransactionBatchFromCheckpoint(v)
 	default:
 		return nil, errors.New("bad message type")
 	}
 }
 
-func DeliveredValue(m Message) value.Value {
+func DeliveredValue(m SingleMessage) value.Value {
 	receiptHash := m.ReceiptHash()
 	receiptVal := big.NewInt(0).SetBytes(receiptHash[:])
 	msg, _ := value.NewTupleFromSlice([]value.Value{
-		value.NewIntValue(m.DeliveredHeight().AsInt()),
-		value.NewIntValue(m.DeliveredTimestamp()),
+		value.NewIntValue(m.deliveredHeight().AsInt()),
+		value.NewIntValue(m.deliveredTimestamp()),
 		value.NewIntValue(receiptVal),
-		m.AsValue(),
+		m.asValue(),
 	})
 	return msg
 }
 
+func AddToPrev(prev value.TupleValue, msg Message) value.TupleValue {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [exported function AddToPrev should have comment or be unexported](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5428340)